src/util/fs.js

```diff
@@ -360,6 +360,11 @@ async function buildActionsForHardlink(
     const {src, dest} = data;
     const onFresh = data.onFresh || noop;
     const onDone = data.onDone || noop;
+    if (files.has(dest)) {
```

Comment:
Oh it does compare inodes for sure, this check just prevents assessing the same path twice, a folder in particular.
We have `A -> B` structure that we want to hardlink to `A1 -> B1`, package-linker passes that modules A1 and B1 need to be hardlinked independently but part of hardlinking is that it is recursive, so files in B1 will be scheduled to be linked twice.

I could not find a more elegant way of doing this especially with bundledDependencies because they don't participate in hoisting.